### PR TITLE
INT-3789 - Fix binary inconsistencies between jenkins.io and central

### PR DESCRIPTION
**Pull request by eduard-tita**
Make sure the same binary ends up in Maven Central and Jenkins.io by building the artifact once and pushing it to both destinations.

Jira: https://issues.sonatype.org/browse/INT-3789

### CODE REVIEW

**Review by tneer — APPROVED** [review]
+1 Looks good to me.

File: Jenkinsfile.sonatype [hunk 1]
```diff
@@ -136,13 +138,9 @@ else {
         runSafely "git push https://${env.GITHUB_API_USERNAME}:${env.GITHUB_API_PASSWORD}@github" +
             ".com/jenkinsci/nexus-platform-plugin.git ${pom.artifactId}-${version}"
       }
-
-      // Reset all changes to local repository made by the Maven release plugin
-      runSafely "git tag -d ${pom.artifactId}-${version}"
-      runSafely 'git clean -f && git reset --hard origin/master'
     }
-    stage('Deploy to Jenkins') {
-      mvn jenkinsMavenConfig(), mvnReleaseCommand
```
Comment (rjackson64840) [inline Jenkinsfile.sonatype:145]: are we using the release command config now?

File: Jenkinsfile.sonatype [hunk 2]
```diff
@@ -96,7 +96,9 @@ else {
         mvnReleaseCommand = "-Darguments=-DskipTests -DreleaseVersion=${version} -DdevelopmentVersion=${pom.version} " +
             "-DpushChanges=false -DlocalCheckout=true -DpreparationGoals=initialize clean release:prepare " +
             "release:perform -B"
-
+        mvnDeployCommand = "-Durl=https://repo.jenkins-ci.org/releases -DrepositoryId=maven.jenkins-ci.org " +
+            "-Dfile=target/nexus-jenkins-plugin.hpi -DgroupId=${pom.groupId} -DartifactId=${pom.artifactId} " +
+            "-Dversion=${version} deploy:deploy-file -B"
```
Comment (rjackson64840) [inline Jenkinsfile.sonatype:101]: do we know the correct version here, before the release is actually created?

**Review by rjackson64840 — APPROVED** [review]
+1 I think the things you added look fine, although I'd recommend some other structural/naming changes in other parts of the script (only affects clarity, not function), but that's outside the scope of what you are doing

File: Jenkinsfile.sonatype (hunk 2 above)
Comment (eduard-tita) [reply Jenkinsfile.sonatype:101]: Yes. It is used in `mvnReleaseCommand` above in a similar way and it works as expected.

**Review by vladt — APPROVED** [review]
+1 Nice work!

**Pull request merged**